pkg/es/config/config.go

```diff
@@ -33,48 +33,75 @@ import (
 	storageMetrics "github.com/jaegertracing/jaeger/storage/spanstore/metrics"
 )
 
+const (
+	IndexPrefixSeparator = "-"
+)
+
+// IndexOptions describes the index format and rollover frequency
+type IndexOptions struct {
+	Priority          int    `mapstructure:"priority"`
+	DateLayout        string `mapstructure:"date_layout"`
+	Shards            int    `mapstructure:"shards"`
+	Replicas          int    `mapstructure:"replicas"`
+	RolloverFrequency string `mapstructure:"rollover_frequency"` // "hour" or "day"
+}
+
+// Indices describes different configuration options for each index type
+type Indices struct {
+	IndexPrefix  IndexPrefix  `mapstructure:"index_prefix"`
+	Spans        IndexOptions `mapstructure:"spans"`
+	Services     IndexOptions `mapstructure:"services"`
+	Dependencies IndexOptions `mapstructure:"dependencies"`
+	Sampling     IndexOptions `mapstructure:"sampling"`
+}
+
+type IndexPrefix string
+
+func (p IndexPrefix) String() string {
```

Comment:
why is this needed?

Reply:
just gives a convenient way to convert to a string instead of having to wrap in a `string()` everytime - i can remove it if you think we don't need it though

Reply:
removed